### PR TITLE
Replace document calls for popup window support

### PR DESCRIPTION
**Pull request by mayank99**
- [x] `getContainer`: Added `ownerDocument` arg and test
- [x] `Modal`: Added `ownerDocument` prop (passed to getContainer)
- [x] `Popover`: Updated `hideOnEsc` plugin to use popper element instead of `document`
- [x] `Table`: added `ownerDocument` (from table ref) to `FilterToggle`
- [ ] `useTheme`: Hook fixed in #97 but none of the components use that arg currently

<details>
<summary>All references to "document" in project</summary>

image
</details>

### CODE REVIEW

File: src/core/utils/Popover.tsx [hunk 1]
```diff
@@ -85,6 +85,9 @@ export const Popover = React.forwardRef((props: PopoverProps, ref) => {
 
 export const hideOnEsc = {
   fn(instance: Instance) {
+    // will always refer to the correct document instance (e.g. in a popout window)
+    const _document = instance.reference?.ownerDocument ?? document;
```
Comment (mayank99) [inline src/core/utils/Popover.tsx:89]: Maybe we need to add a util like this: https://github.com/mui-org/material-ui/blob/next/packages/material-ui-utils/src/ownerDocument.ts

**Comment by mayank99** [comment]
This is ready for review now. Check PR description.

I don't think we need to update `useTheme` call in all components yet (it could get complicated with refs and stuff, might need a util).

**Comment by mayank99** [comment]
> About table filter, can we get `ownerDocument` of table ref and is that?

Done. But react-table still has some `document` references, which we can't do anything about.

https://github.com/tannerlinsley/react-table/blob/master/src/plugin-hooks/useResizeColumns.js#L105
https://github.com/tannerlinsley/react-table/blob/master/src/plugin-hooks/useGridLayout.js#L82

File: src/core/Table/Table.tsx [hunk 2]
```diff
@@ -349,7 +352,10 @@ export const Table = <
                   <div {...columnProps} key={columnProps.key}>
                     {column.render('Header')}
                     {!isLoading && data.length != 0 && (
-                      <FilterToggle column={column} />
+                      <FilterToggle
+                        column={column}
+                        ownerDocument={tableRef.current?.ownerDocument}
```
Comment (mayank99) [inline src/core/Table/Table.tsx:357]: Actually, this won't work because it's passed at the same time as the ref as initialized so it's null

Comment (mayank99) [reply src/core/Table/Table.tsx:357]: Fixed and added test

**Comment by bentleyvk** [comment]
> > About table filter, can we get `ownerDocument` of table ref and is that?
> 
> Done. But react-table still has some `document` references, which we can't do anything about.
> 
> https://github.com/tannerlinsley/react-table/blob/master/src/plugin-hooks/useResizeColumns.js#L105
> https://github.com/tannerlinsley/react-table/blob/master/src/plugin-hooks/useGridLayout.js#L82

Yeah but we are not using them right now.

**Comment by bentleyvk** [comment]
Have you tested all these changes with a popup? 😀

**Comment by mayank99** [comment]
> Have you tested all these changes with a popup? 😀

Did now, and it all works! (Took me a while to get linked cra working locally and I had to copy styles from main window into popup)
<details>
<summary>copyStyles</summary>

```ts
function copyStyles(src: Document, dest: Document) {
  Array.from(src.styleSheets).forEach(styleSheet => {
    styleSheet && dest.head.appendChild(styleSheet!.ownerNode!.cloneNode(true))
  })
}
```
</details>

I tested linking a build from main branch as well, to compare!
- [x] `Table` filter dropdown is broken on main (and works here)
- [x] `DropdownMenu` escape key doesn't work on main (and works here)
- [x] `Modal`: Passing `ownerDocument` works
- [x] `ThemeProvider`: Passing `ownerDocument` works

**Review by veekeys — APPROVED** [review]

**Pull request merged**